### DISCUSSION TITLE
XS4ALL VDSL PPPoE Passthrough

### DISCUSSION THREAD

**Discussion by Jordenjj** [fabianishere/udm-iptv#81] [Q&A] [answered]
Hello,
Has anyone gotten this to work with XS4ALL and VDSL PPPoE Passthrough? It doesn't work for me, and the weird thing is I only get internet using no VLAN ID. VLAN 6 doesn't work.

**Comment by fabianishere** [comment]
I expect your VDSL modem is only passing through the PPPoE-encapsulated traffic from VLAN 6. That means that it will wrap the packets you send to it into a PPPoE packet over VLAN 6.

**Reply by Jordenjj** [reply]
Does this mean IPTV won't be possible using this method?

**Reply by fabianishere** [reply]
If my assumptions are right, probably not, unless you manage to also forward traffic from VLAN 4.

**Reply by fabianishere** [reply]
Can't you configure your modem to disable PPPoE passthrough and just forward the tagged VLANs?

**Reply by Jordenjj** [reply]
> Can't you configure your modem to disable PPPoE passthrough and just forward the tagged VLANs?

Don't think so. Fritzboxes are generally great little devices except for when it comes to letting other stuff manage connections

**Reply by fabianishere** [reply]
Does IPTV work when connected directly to your Fritzbox? If so, you can probably set `IPTV_WAN_VLAN` to 0 to get it working.

**Reply by Jordenjj** [reply]
What do you mean with directly connected to the Fritzbox? The UDM Receiving DHCP? Yes TV works with IGMP snooping turned on when the UDM gets its ip from DHCP. Even with double nat. It doesnt seem stable though. I'll give it a shot soon.

'Preciate you helping everyone out like this

Edit: its still really unstable but it seems to sometimes get signal in dhcp mode

**Reply by Jordenjj** [reply]
Also, I HAVE to set VLAN ID 6 on the modem passthrough settings. Otherwise it doesn't work. So I think only VLAN 6 gets passed through...
<img width="574" alt="Screenshot 2022-03-01 at 20 55 13" src="https://user-images.githubusercontent.com/14220294/156239149-123d9f19-730f-4949-a900-59302f9353c5.png">

**Reply by fabianishere** [reply]
What Fritzbox model are you using?

**Reply by Jordenjj** [reply]
> What Fritzbox model are you using?

7583

**Reply by fabianishere** [reply]
By the way, what do you mean by it is getting signal in DHCP mode?

You configure the WAN port on the UDM to use DHCP? And getting signal means the STB is showing image?

In that case, you can probably get away with running double NAT and setting `IPTV_WAN_VLAN` to 0. The last is necessary to properly route IPTV traffic between the UDM and the Fritzbox.

**Reply by fabianishere** [reply]
Also, I recommend moving your STBs into a separate VLAN to reduce interference with other devices.

**Reply by Jordenjj** [reply]
When I get internet by connecting the WAN to the Fritz in DHCP mode I get image, but it is really unstable.
I get no image when the UDM gets internet through PPPoE. I tried VLAN 0 but also doesn't work.

**Reply by fabianishere** [reply]
In your case, the Fritzbox already decapsulates the PPPoE traffic into an untagged LAN. Thus, DHCP mode for the WAN port on the UDM is the right choice.

This guide assumes there is no intermediate device between your ISP and the UDM doing any packet processing.

**Reply by Jordenjj** [reply]
Then I guess I'm screwed! Either no TV or TV that crashes every 5 seconds.

Thank you for helping, I guess I'll look to return the UDM.

**Reply by fabianishere** [reply]
You should be able to get TV working on the UDM if the STB also works directly via the Fritzbox.

To verify this, you’ll have to plug the STB directly in the Fritzbox and see whether the TV works okay.

In that case, setting `IPTV_WAN_VLAN` to 0 should make it work on the UDM.

If IPTV does not work when connected to the Fritzbox, the issue lies with the Fritzbox.

**Reply by Jordenjj** [reply]
IPTV works on the FritzBox. IPTV_WAN_VLAN to 0 unfortunately doesn't...

**Reply by Jordenjj** [reply]
<img width="581" alt="Screenshot 2022-03-01 at 22 20 02" src="https://user-images.githubusercontent.com/14220294/156250676-5e81fb92-5610-4967-9714-058efb4da48a.png">

Im not sure if this has anything to do with it

**Reply by fabianishere** [reply]
Could you share the output of `udm-iptv-diag`?

**Reply by Jordenjj** [reply]
> Could you share the output of `udm-iptv-diag`?
```
Please share the following output with the developers:
=== Configuration ===
WAN Interface: eth8
WAN Ranges: 213.75.0.0/16 217.166.0.0/16
WAN VLAN: 0 (dev iptv)
WAN DHCP options: -O staticroutes -V IPTV_RG
LAN Interfaces: br0
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: false
=== IP Link and Route ===
=== Service Logs ===
Mar 01 23:01:56 ubnt systemd[1]: Stopped IPTV support for the UniFi Dream Machine.
Mar 01 23:01:56 ubnt systemd[1]: udm-iptv.service: Consumed 87ms CPU time.
Mar 01 23:01:56 ubnt systemd[1]: Started IPTV support for the UniFi Dream Machine.
Mar 01 23:01:56 ubnt udm-iptv[19939]: NATing IPTV network ranges (if necessary)
Mar 01 23:01:56 ubnt udm-iptv[19939]: Setting up igmpproxy
Mar 01 23:01:56 ubnt udm-iptv[19939]: Starting igmpproxy
Mar 01 23:01:56 ubnt igmpproxy[19939]: There must be at least 1 Vif as upstream.
Mar 01 23:01:56 ubnt systemd[1]: udm-iptv.service: Main process exited, code=exited, status=255/EXCEPTION
Mar 01 23:01:56 ubnt systemd[1]: udm-iptv.service: Failed with result 'exit-code'.
Mar 01 23:01:56 ubnt systemd[1]: udm-iptv.service: Consumed 80ms CPU time.
root@ubnt:/etc# 

```

**Reply by fabianishere** [reply]
This error means your `eth8` interface does not have an IP address. Is that correct?

If you configure the WAN port in DHCP mode (VLAN 0), it should have obtained an IP address.

**Reply by Jordenjj** [reply]
> This error means your `eth8` interface does not have an IP address. Is that correct?
> 
> If you configure the WAN port in DHCP mode (VLAN 0), it should have obtained an IP address.

Yeah sorry I was messing around with the settings. Running DHCP now.

```
Please share the following output with the developers:
=== Configuration ===
WAN Interface: eth8
WAN Ranges: 213.75.0.0/16 217.166.0.0/16
WAN VLAN: 0 (dev iptv)
WAN DHCP options: -O staticroutes -V IPTV_RG
LAN Interfaces: br0
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: false
=== IP Link and Route ===
4: eth8: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq qlen 1000
    inet 192.168.178.20/24 scope global dynamic eth8
       valid_lft 863859sec preferred_lft 863859sec
192.168.178.0/24 dev eth8  src 192.168.178.20 
=== Service Logs ===
Mar 01 23:04:02 ubnt systemd[1]: udm-iptv.service: Failed with result 'exit-code'.
Mar 01 23:04:02 ubnt systemd[1]: udm-iptv.service: Consumed 82ms CPU time.
Mar 01 23:04:08 ubnt systemd[1]: udm-iptv.service: Service RestartSec=5s expired, scheduling restart.
Mar 01 23:04:08 ubnt systemd[1]: udm-iptv.service: Scheduled restart job, restart counter is at 1122.
Mar 01 23:04:08 ubnt systemd[1]: Stopped IPTV support for the UniFi Dream Machine.
Mar 01 23:04:08 ubnt systemd[1]: udm-iptv.service: Consumed 82ms CPU time.
Mar 01 23:04:08 ubnt systemd[1]: Started IPTV support for the UniFi Dream Machine.
Mar 01 23:04:08 ubnt udm-iptv[24415]: NATing IPTV network ranges (if necessary)
Mar 01 23:04:08 ubnt udm-iptv[24415]: Setting up igmpproxy
Mar 01 23:04:08 ubnt udm-iptv[24415]: Starting igmpproxy
root@ubnt:/etc# 

```

**Reply by fabianishere** [reply]
It seems `igmpproxy` is running correctly now, so I except IPTV to work.

**Reply by Jordenjj** [reply]
> It seems `igmpproxy` is running correctly now, so I except IPTV to work.

Channels only work for about 1-3 seconds, then they freeze and get an error saying the STB disconnected from modem. Switching channels makes them work for a couple seconds again

**Reply by fabianishere** [reply]
Could you try if setting `IPTV_WAN_RANGES` to `0.0.0.0/0` fixes the issue?

**Reply by Jordenjj** [reply]
> Could you try if setting `IPTV_WAN_RANGES` to `0.0.0.0/0` fixes the issue?

No improvement unfortunately.

**Reply by fabianishere** [reply]
You restarted the service, right? 
If so, the UDM is somehow blocking the non-multicast UDP IPTV packets.

**Reply by Jordenjj** [reply]
> You restarted the service, right?
Yes using systemctl restart udm-iptv
> If so, the UDM is somehow blocking the non-multicast UDP IPTV packets.
Im not sure what that means ;) Not too literate on this

**Reply by fabianishere** [reply]
To diagnose the issue, you’ll need to make a packet trace of both sides of the network when starting the STB.

**Reply by Jordenjj** [reply]
> To diagnose the issue, you’ll need to make a packet trace of both sides of the network when starting the STB.

Im not sure what that means... But I found something out: I can rewind shows and play them, and they work! For some reason... Live TV doesnt

**Comment by Jordenjj** [comment]
My ISP says to also put in these rules. Does this tell you anything?
55 (parameter-rquest-list), bevat minstens 1, 3, 28 en 121

[XS4ALL IPTV Routed](https://www.xs4all.nl/klant/instellingen-routed-mode-televisie-voor-ander-modem/)

**Reply by fabianishere** [reply]
That’s only relevant for the Fritzbox configuration (which is already working).

**Reply by Jordenjj** [reply]
> That’s only relevant for the Fritzbox configuration (which is already working).

Ah. I understand.

**Comment by Jordenjj** [comment, marked as answer]
I ordered a DrayTek Vigor 167 to use a true DSL bridge mode modem
